dynamo tracing perf: don't unnecessarily call getframeinfo on the hot path: 47.26 -> 37.66

aorenste opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #143061
* #143070
* #143069
* #143063
* #143062
* #143057
* #143068
* #143067
* #143065
* #143064
* #143060
* #143059
* #143058
* #143056
* __->__ #143066

See #143056 for overall docs.

This PR: Stop using `getframeinfo()` when we only care about the function name
and throw the rest away.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

aorenste committed:
don't unnecessarily call getframeinfo on the hot path: 47.26 -> 37.66

See #143056 for overall docs.

This PR: Stop using `getframeinfo()` when we only care about the function name
and throw the rest away.

[ghstack-poisoned]

aorenste committed:
Update on "don't unnecessarily call getframeinfo on the hot path: 47.26 -> 37.66"


See #143056 for overall docs.

This PR: Stop using `getframeinfo()` when we only care about the function name
and throw the rest away.

[ghstack-poisoned]

jansel reviewed (approved):
The win on this one is huge, maybe we should try to land it first.

aorenste committed:
Update on "dynamo tracing perf: don't unnecessarily call getframeinfo on the hot path: 47.26 -> 37.66"


See #143056 for overall docs.

This PR: Stop using `getframeinfo()` when we only care about the function name
and throw the rest away.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

aorenste committed:
Update on "dynamo tracing perf: don't unnecessarily call getframeinfo on the hot path: 47.26 -> 37.66"


See #143056 for overall docs.

This PR: Stop using `getframeinfo()` when we only care about the function name
and throw the rest away.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

aorenste committed:
Update on "dynamo tracing perf: don't unnecessarily call getframeinfo on the hot path: 47.26 -> 37.66"


See #143056 for overall docs.

This PR: Stop using `getframeinfo()` when we only care about the function name
and throw the rest away.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]
